### PR TITLE
Improve segment pool test coverage

### LINKED ISSUE #356: Improve SegmentPool test coverage
Currently, there are a few naïve tests checking how `SegmentPool` works on JVM.
Unfortunately, these tests can't catch up with increasing complexity of the underlying implementation.

### PR DESCRIPTION
**Pull request by fzhinkin**
Closes #356

### CODE REVIEW

File: core/common/src/Segment.kt [hunk 1]
```diff
@@ -20,6 +20,7 @@
  */
 package kotlinx.io
 
+import kotlin.concurrent.Volatile
```
Comment (shanshin) [inline core/common/src/Segment.kt:23]: Unused?

File: core/jvm/test/PoolingTest.kt [hunk 2]
```diff
@@ -40,4 +45,107 @@ class PoolingTest {
...
+        }
+        segments2.forEach(SegmentPool::recycle)
+
+        assertEquals(segments, segments2)
+    }
+
+    @Test
+    fun contendedUseTest() {
+        val threadsCount = Runtime.getRuntime().availableProcessors()
+        val segmentsPerThread = SegmentPool.SECOND_LEVEL_POOL_TOTAL_SIZE / Segment.SIZE / threadsCount
```
Comment (shanshin) [inline core/jvm/test/PoolingTest.kt:82]: It turns out that we only take into account fully filled segments?

File: core/jvm/test/PoolingTest.kt [hunk 3]
```diff
@@ -40,4 +45,107 @@ class PoolingTest {
...
+                        observedSegments[seg] = seg
+                    }
+                    segments.forEach { SegmentPool.recycle(it) }
+                }
+            }
+        }
+        pool.shutdown()
+        pool.awaitTermination(Long.MAX_VALUE, TimeUnit.SECONDS)
+
+        val maxPoolSize =
```
Comment (shanshin) [inline core/jvm/test/PoolingTest.kt:102]: Should we try to limit the number of segments more precisely?
E.g. use `(SegmentPool.SECOND_LEVEL_POOL_TOTAL_SIZE + SegmentPool.MAX_SIZE * threadsCount) / Segment.SIZE`

File: core/jvm/test/PoolingTest.kt [hunk 4]
```diff
@@ -40,4 +45,107 @@ class PoolingTest {
...
+            (SegmentPool.SECOND_LEVEL_POOL_TOTAL_SIZE + SegmentPool.MAX_SIZE * SegmentPool.HASH_BUCKET_COUNT) / Segment.SIZE
```
Comment (shanshin) [inline core/jvm/test/PoolingTest.kt:148]: same as above

File: core/jvm/test/PoolingTest.kt (hunk 2 above)
Comment (fzhinkin) [reply core/jvm/test/PoolingTest.kt:82]: Pool size reflects maximum number of segments it'll hold once they returned back into pool. When segments returned, they are all empty, so their whole size contributes into the pool size.

File: core/jvm/test/PoolingTest.kt (hunk 3 above)
Comment (fzhinkin) [reply core/jvm/test/PoolingTest.kt:104]: Instead, I increased the number of threads to make sure that it'll exceed the number of buckets

**Review by shanshin — APPROVED** [review]

**Pull request merged**